src/bin/main.rs

```diff
@@ -305,20 +332,35 @@ fn main() {
         };
     }
 
+    let mut track_copy_dest = HashMap::new();
     if let Some(code_dir) = code_dir {
         if let Some(code_files) = code_files {
-            for code_file in &code_files {
-                eprintln!("Copying code file {:?}", code_file);
+            for (code_file, code_out_path) in &code_files {
+                if track_copy_dest.contains_key(code_out_path) {
+                    eprintln!(
+                        "ERROR: Attempted to copy multiple code files to {:?}: from {:?} and {:?}",
+                        code_out_path, track_copy_dest[code_out_path], code_file
+                    );
+                    return;
+                }
+                track_copy_dest.insert(code_out_path, code_file);
+
                 let mut file_path = code_dir.clone();
```

Comment:
Concerning efficiency, I wonder if it would make sense to delay resolving the glob patterns and doing the path replacement until here as the glob patterns could make the expanded list very large with the target paths adding memory overhead.

Reply:
The problem here is that the paths may be different. So, using the glob "**/*.txt" may result in a different first component.

Reply:
But the expanded code files is essentially a `flat_map` over the code files. I am merely proposing to move this inner loop here without changing the logic at all (instead of collecting into a vector and having a single loop here).

Reply:
Ok, so I probably don't understand what you mean...

There is one replacement "pattern" per glob pattern. I iterate over globs and replacements "in parallel", so I always get a (glob, replacement) pair. I then get all the files for the glob, and do the replacement (using the same replacement "pattern") on every of those files.

Could you try to explain your suggestion in such a verbal way?

Reply:
> Could you try to explain your suggestion in such a verbal way?

That part does not change at all. I am just suggesting to avoid collecting all the source-destination pairs in a vector before iterating over the entries of this vector to copy source to destination.

Instead of
```rust
for (code_file, code_out_path) in &code_files { /* copy code_file to code_out_path */ }
```
you could do something like
```rust
for (idx, code_file) in paths.code_files.iter().enumerate() {
  let code_path = paths.code_paths.as_ref().map(|code_paths| code_paths[idx]);

  for code_file in PathUtil::list_files(code_file) {
    let out_path = code_path.map_or(code_file, |code_path| modify_path(code_file, code_path));

    /* copy code_file to out_path */
  }
}
```

Reply:
Ah, thanks!

Reply:
Implemented the suggestion, also for doc files. However, it was quite some fight against the borrow checker, and I probably have one or more "unnecessary" `clone`s there.